domains/middleware/authorization.go

```diff
@@ -134,6 +140,69 @@ func (am *authorizationMiddleware) ListDomains(ctx context.Context, session auth
 	return am.svc.ListDomains(ctx, session, page)
 }
 
+func (am *authorizationMiddleware) SendInvitation(ctx context.Context, session authn.Session, invitation domains.Invitation) (err error) {
+	session.DomainUserID = auth.EncodeDomainUserID(session.DomainID, session.UserID)
+	domainUserId := auth.EncodeDomainUserID(invitation.DomainID, invitation.UserID)
+	if err := am.extAuthorize(ctx, domainUserId, policies.MembershipPermission, policies.DomainType, invitation.DomainID); err == nil {
+		// return error if the user is already a member of the domain
+		return errors.Wrap(svcerr.ErrConflict, ErrMemberExist)
+	}
+
+	if err := am.checkAdmin(ctx, session); err != nil {
+		return err
+	}
+
+	return am.svc.SendInvitation(ctx, session, invitation)
+}
+
+func (am *authorizationMiddleware) ViewInvitation(ctx context.Context, session authn.Session, userID, domain string) (invitation domains.Invitation, err error) {
+	session.DomainUserID = auth.EncodeDomainUserID(session.DomainID, session.UserID)
+	if session.UserID != userID {
+		if err := am.checkAdmin(ctx, session); err != nil {
+			return domains.Invitation{}, err
+		}
+	}
+
+	return am.svc.ViewInvitation(ctx, session, userID, domain)
+}
+
+func (am *authorizationMiddleware) ListInvitations(ctx context.Context, session authn.Session, page domains.InvitationPageMeta) (invs domains.InvitationPage, err error) {
+	session.DomainUserID = auth.EncodeDomainUserID(session.DomainID, session.UserID)
+	if err := am.extAuthorize(ctx, session.UserID, policies.AdminPermission, policies.PlatformType, policies.SuperMQObject); err == nil {
+		session.SuperAdmin = true
+	}
+
+	if !session.SuperAdmin {
+		switch {
+		case page.DomainID != "":
+			if err := am.extAuthorize(ctx, session.DomainUserID, policies.AdminPermission, policies.DomainType, page.DomainID); err != nil {
+				return domains.InvitationPage{}, err
+			}
+		default:
+			page.InvitedByOrUserID = session.UserID
+		}
+	}
+
+	return am.svc.ListInvitations(ctx, session, page)
+}
```

Comment:
@felixgateru 
Session will get domain ID from the URL, we have removed domain login, so session will not get domain ID from token. 
So please check and add domain ID in URL like 
`/<domainID>/invitations`  
POST - Will add invitations to domain 
GET - will list pending in domain 
DELETE will delete invitations in domain 

And for invitee user 
We can provide endpoint like 
GET `/invitations`
POST `/invitations/accept`
POST `/invitations/reject`